clients/include/rocblas_math.hpp

```diff
@@ -20,7 +20,14 @@ inline __host__ rocblas_half float_to_half(float val)
 }
 
 // Helper routine to convert halfs into their floats equivalent; uses F16C instructions
-inline __host__ float half_to_float(rocblas_half val)
+template <typename T>
+inline __host__ float half_to_float(T val)
+{
```

Comment:
`half_to_float` should not be generalized like this. It merely defers a compile-time error to a run-time error.

This is probably being done because the current templated code calls `half_to_float` in a part of the code guarded by `if` statements but always compiled in regardless of the types, which can lead to compile-time errors if complex types are used.

In C++17 we can use `constexpr if` to produce conditional code and avoid compilation errors, but since we aren't using C++17 yet, we have to use SFINAE or other techniques.